idaes/core/util/model_diagnostics.py

```diff
@@ -4359,17 +4448,24 @@ def __init__(
         term_mismatch_tolerance: float = 1e6,
         term_cancellation_tolerance: float = 1e-4,
         term_zero_tolerance: float = 1e-10,
+        max_canceling_terms: int = 4,
+        max_cancellations_per_node: int = 5,
```

Comment:
Why do these have default values in the object when we're also giving them default values in the `ConfigBlock`?

Reply:
Different objects - the is the visitor class which exists outside the `DiagnosticsToolbox` class and so can be used separately if desired. Thus, the visitor has the ability to take these as arguments and thus has default values for them.